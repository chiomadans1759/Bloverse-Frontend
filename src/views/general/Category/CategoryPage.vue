<template>
  <main id="category-page">
    <div class="main">
        <navbar :showBackArrow="showBackArrow" :showNavigations="showNavigations" customClass="back-nav"/>
      <div class="container">
        <div class="row d-flex justify-content-center align-items-center">
          <div class="col-sm-7 col-lg-6 pt-5">
            <h5 class="pb-1">Select Your Interest</h5>
            <p class="select-t">Selected {{consumers.selectedCategories.length}} categories</p>
            <div>
              <choose-catergory></choose-catergory>
            </div>
          </div>
        </div>
        <div class="card category-footer py-2 rounded-0">
          <div class="container">
            <div class="row d-flex justify-content-center">
              <div class="col-sm-7 col-lg-6 d-flex justify-content-between">
                <button class="btn btn-link btn-sm" type="button">Skip</button>
                <button class="btn btn-sm btn-primary">Save</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</template>

<script>
import ChooseCatergory from "@/components/ChooseCategory/ChooseCategory.vue";
import Navbar from "@/components/Navbar/Navbar.vue";
import { mapState } from "vuex";

export default {
  components: { ChooseCatergory, Navbar },
  data() {
    return {
      showNavigations: false,
      showBackArrow: true
    };
  },
  computed: {
    ...mapState(["consumers"])
  }
};
</script>

<style lang="scss" scoped>
@import "./CategoryPage";
</style>
