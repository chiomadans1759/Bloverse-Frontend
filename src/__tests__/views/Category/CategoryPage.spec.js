import { mount, createLocalVue, RouterLinkStub  } from '@vue/test-utils';
import Category from '@/views/general/Category/CategoryPage.vue';
import categoryData from '@/fixtures/category.js';

import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('ChooseCategory Component', () => {
  let actions
  let store

  beforeEach(() => {
    actions = {
      addToCategories: jest.fn()
    }
    store = new Vuex.Store({
      state: {
        consumers: {
          selectedCategories: [...categoryData],
        }
      },
      actions
    })
  });

  it('dispatches "actionInput" when input event value is "input"', () => {
    const wrapper = mount(Category, { 
      store, 
      localVue,
      stubs: {
        RouterLink: RouterLinkStub
      }
    });

    wrapper.find('.overlay').trigger('click');
    expect(actions.addToCategories).toHaveBeenCalled();
    expect(wrapper.element).toMatchSnapshot();
  })
});
