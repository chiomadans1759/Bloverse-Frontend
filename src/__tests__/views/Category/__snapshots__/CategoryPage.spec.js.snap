// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Page should render correctly 1`] = `
<main 
  id="category-page"
  >
    <go-back-header>
    </go-back-header>

    <div 
      class="main"
    >
    <div 
      class="container"
    >
      <div 
        class="row d-flex justify-content-center align-items-center"
      >
        <div 
          class="col-sm-7 col-lg-6 pt-5"
        >
          <h5 class="pb-1">
            Select Your Interest
          </h5>
         <p 
          class="select-t"
         >
          Selected {{consumers.selectedCategories.length}} categories
          </p>
          <div>
           <choose-catergory>
           </choose-catergory>
          </div>
        </div>
      </div>
      <div 
        class="card category-footer py-2 rounded-0"
      >
        <div 
          class="container"
        >
            <div 
              class="row d-flex justify-content-center"
            >
              <div 
                class="col-sm-7 col-lg-6 d-flex justify-content-between"
              >
                <button 
                  class="btn btn-link btn-sm" type="button"
                > 
                    Skip
                </button>
                <button 
                  class="btn btn-sm btn-primary"
                >
                  Save
                </button>
              </div>
            </div>
          </div>
        </div>
    </div>
    </div>
  </main>
`;